README.md

```diff
@@ -37,185 +41,198 @@ Simple parsing for node.js.
+```js
+expression ->
+      number "+" number
+    | number "-" number
+    | number "*" number
+    | number "/" number
+```
 
-A nonterminal can have multiple expansions, separated by vertical bars (`|`):
+A special kind of rule is an **epsilon rule**. It matches nothing and is written as `null`. The following nonterminal matches zero or more `cow`s in a row:
 
-    expression ->
-          number "+" number
-        | number "-" number
-        | number "*" number
-        | number "/" number
+```js
+a -> null
+    | a "cow"
+```
 
-The parser tries to parse the first nonterminal that you define in a file.
-However, you can (and should!) introduce more nonterminals as "helpers". In
-this example, we would have to define the expansion of `number`.
+Keep in mind that nearley syntax is not sensitive to formatting. You may keep rules on the same line: `foo -> bar | qux`.
 
+### Comments
+
+Everything starting from `#` to the end of a line is ignored as a comment:
+
+```ini
+expression -> number "+" number # sum of two numbers
+```
 
 ### Postprocessors
 
-Each meaning (called a *production rule*) can have a postprocessing function,
-that can format the data in a way that you would like:
+By default, nearley wraps everything matched by a rule into an array. `rule -> "foo" "bar"` gives `["foo", "bar"]`.
+Most of the time, you need to process that data in some way: filter out unnecessary tokens, transform into an object, etc.
+
+Each rule can have a *postprocessor* - a JavaScript function that transforms the array and returns whatever you want to get instead. Postprocessors are wrapped in `{% %}`:
 
 ```js
 expression -> number "+" number {%
-    function (data, location, reject) {
-        return ["sum", data[0], data[2]];
-    }
+    (data, location, reject) => ({
+        type: "sum",
+        args: [data[0], data[2]]
+    })
 %}
 ```
 
-`data` is an array whose elements match the nonterminals in order. The
-postprocessor `id` returns the first token in the match (literally
-`function(data) {return data[0];}`).
+The rule above will parse `5+10` into `{ type: "sum", args: [5, 10] }`.
+
+The postprocessor can be any function. It will be passed three arguments:
 
-`location` is the index at which that rule was found. Retaining this
-information in a syntax tree is useful if you're writing an interpreter and
-want to give fancy error messages for runtime errors.
+- `data: Array` - array with the parsed parts of the rule. It will always be an array, even if there's only one part. If a rule contains nonterminals with their own postprocessors, the respective parts will already be transformed.
+- `location: number` - the index (zero-based) at which the rule match starts. It's useful to retain this information in the syntax tree if you're writing an interpreter.
+- `reject: Object` - return this object to signal that this rule doesn't actually match. This allows you to restrict or conditionally support language features.
```

Comment:
Also, we should probably warn people to avoid `reject` if possible. @Hardmath123